### PR TITLE
Add a tutorial for ssh and socks proxy usage

### PR DESCRIPTION
**Pull request by jimbobhickville**
Is this explanatory enough?  Anything I missed?

### CODE REVIEW

File: docs/ssh.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
+Using SSH to Access Your Cluster
+===========
```
Comment (thesquelched) [inline docs/ssh.rst]: I think this will break parsing if you don't extend it the length of the header

File: docs/ssh.rst [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Once you have a cluster, many of the tools in the ecosystem require command line access to your
+cluster in order to run them.  We provide SSH access to all nodes in your cluster, as well as the
+ability to set up a SOCKS proxy over SSH so you can load internal cluster web pages in your
+browser.
+
+Setting up your SSH credentials
+--------------
```
Comment (thesquelched) [inline docs/ssh.rst]: Same with this header

Comment (thesquelched) [inline docs/ssh.rst]: And you can find the rest

File: docs/ssh.rst [hunk 3]
```diff
@@ -0,0 +1,131 @@
+Using SSH to Access Your Cluster
+===========
+
+Once you have a cluster, many of the tools in the ecosystem require command line access to your
+cluster in order to run them.  We provide SSH access to all nodes in your cluster, as well as the
```
Comment (thesquelched) [inline docs/ssh.rst]: non-Ambari nodes?

File: docs/ssh.rst (hunk 1 above)
Comment (jimbobhickville) [inline docs/ssh.rst]: It has to be the same length?  I had just copied it from another file.  I don't understand rst :(

Comment (thesquelched) [inline docs/ssh.rst]: I ran into this issue when I was updating the PyPi project page.  You can run `make -C docs clean html` and then open up `docs/_build/html/index.html` to make sure.

Comment (jimbobhickville) [inline docs/ssh.rst]: It's working fine, however the line breaks in the indented blocks seems to mean nothing :(

File: docs/ssh.rst [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+If you haven't defined any ssh credentials, when you create a cluster using lavaclient, it
+will attempt to use your current username and ssh public key to provide access to your cluster.
+
+    $ NODE_GROUPS='slave(flavor_id=hadoop1-7,count=3)'
+    $ lavaclient clusters create --node-groups $NODE_GROUPS my-cluster HADOOP_HDP2_3
+    You have not uploaded any SSH key credentials; do you want to upload
+    $HOME/.ssh/id_rsa.pub now?
+
+If you want to specify a different ssh key, you can preload your ssh keys using the credentials
+commands:
```
Comment (thesquelched) [inline docs/ssh.rst]: I thought you needed two colons in order to get code highlighting

File: docs/ssh.rst [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+    $ lavaclient credentials create_ssh_key my-ssh-key ~/.ssh/id_rsa.pub
```
Comment (thesquelched) [inline docs/ssh.rst]: `s/lavaclient/lava`

File: docs/ssh.rst [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+The IP addresses of your cluster nodes are visible by getting a list of nodes in your cluster:
+
+    $ lavaclient clusters nodes <cluster-id>
+
+
+SSH Access
+--------------
+
+lavaclient has a shortcut for ssh'ing into a node in your cluster.  By default, it will pick a
```
Comment (thesquelched) [inline docs/ssh.rst]: Aside from the `lavaclient` -> `lava` thing, you should either change this to `The lava client`, or put backticks around `lava`

File: docs/ssh.rst [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    Last login: Thu Jun 18 18:25:50 2015 from some.host
+    [my-username@gateway-1 ~]$
+
+If you want a specific host, you can use the node name to go to the host of your choice:
+
+    $ lava clusters ssh --node-name master-1 <cluster-id>
+    Last login: Thu Jun 18 18:25:50 2015 from some.host
+    [my-username@master-1 ~]$
+
+If you need custom ssh options, like you are using a different ssh keypair than the default, you
```
Comment (thesquelched) [inline docs/ssh.rst]: `like` sounds too conversational; try `e.g.` or `for example`?

File: docs/ssh.rst [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+can supply a custom ssh command to be used:
```
Comment (thesquelched) [inline docs/ssh.rst]: Either put backticks around `ssh` or use `SSH`

File: docs/ssh.rst [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+    $ lava clusters ssh --node-name gateway-1
+    Last login: Thu Jun 18 18:25:50 2015 from some.host
+    [my-username@gateway-1 ~]$
+
+
+SOCKS Proxy
+--------------
+
+Many of the services in your cluster provide web pages to get status information about that
```
Comment (thesquelched) [inline docs/ssh.rst]: `provide web pages for status information.`

File: docs/ssh.rst [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+service.  Because these pages are not locked down with authenticated access, we firewall off your
```
Comment (thesquelched) [inline docs/ssh.rst]: I prefer passive voice here: `public access is disabled using a firewall`

File: docs/ssh.rst [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+cluster to deny public access to them.  In order for you to access them, you need to create what
```
Comment (thesquelched) [inline docs/ssh.rst]: The `what is called` is extraneous; maybe just make `SOCKS proxy` a link to a wikipedia page or something.

Comment (thesquelched) [inline docs/ssh.rst]: Also `Proxy` isn't a proper noun

File: docs/ssh.rst [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+is called a SOCKS Proxy to a node in your cluster, at which point you can configure your web
+browser to use this proxy and access the web pages as if you are within the cluster itself.  It
+is a SOCKS 5 Proxy, so make sure not to configure your browser using SOCKS 4.
```
Comment (thesquelched) [inline docs/ssh.rst]: Change this last sentence to a note:

```
note::

    Make sure to configure your browser to use a SOCKS 5 proxy, not SOCKS 4
```

Comment (thesquelched) [inline docs/ssh.rst]: Probably after the example

File: docs/ssh.rst [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+
+    $ lava clusters ssh_proxy <cluster_id>
+    Starting SOCKS proxy via node gateway-1 (xxx.xxx.xxx.xxx)
+    Successfully created SOCKS proxy on localhost:12345
+    Use Ctrl-C to stop proxy
+
+Until you hit Ctrl-C to kill the proxy, you can point your browser to use the proxy address
+localhost:12345 at which point any page loads will be proxied through your cluster.
```
Comment (thesquelched) [inline docs/ssh.rst]: Backticks around `localhost:12345`

Comment (thesquelched) [inline docs/ssh.rst]: And a comma afterward

File: docs/ssh.rst [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+
+*NOTE* While the proxy is running, you probably want to avoid watching Netflix or Youtube, as
```
Comment (thesquelched) [inline docs/ssh.rst]: Use sphinx notes:

```
note::

    While the proxy is running...
```

File: docs/ssh.rst [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+that traffic will be proxied through your cluster and you will pay for the bandwidth used.  We
```
Comment (thesquelched) [inline docs/ssh.rst]: In general, I don't like the use of `we` here.  I think the imperative works better in documentation, e.g. `Try using a browser plugin...`

Comment (thesquelched) [inline docs/ssh.rst]: And by "here", I mean this PR and all docs in general

File: docs/ssh.rst [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+suggest using a browser plugin that lets you easily turn proxies on and off so you only use it
+when you need to access your cluster.
+
+All of your cluster nodes have a local hostname associated with them, so once you have the proxy
```
Comment (thesquelched) [inline docs/ssh.rst]: `have an associated local hostname`

File: docs/ssh.rst [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+running, the local cluster DNS names should resolve for you.
```
Comment (thesquelched) [inline docs/ssh.rst]: `will resolve.`

File: docs/ssh.rst [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+
+    http://master-1.local:50070
```
Comment (thesquelched) [inline docs/ssh.rst]: It's soft of awkward to have this floating here with no explanation; maybe do it inline in parentheses after the `local cluster DNS names` part earlier?

**Comment by jimbobhickville** [comment]
@thesquelched I think I addressed all of your concerns.  I also redid some of the organization a little.

File: docs/access.rst [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+    Use Ctrl-C to stop proxy
+
+.. note::
+
+    Make sure you configure your browser to use a SOCKS 5 proxy, not SOCKS 4. Each browser has a
+    different method for configuring a proxy, so refer to your browser documentation if you don't
+    know how to configure a proxy.
+
+Until you hit Ctrl-C to kill the proxy, you can point your browser to use the proxy address
+`localhost:12345`, at which point any page loads will be proxied through your cluster.
```
Comment (thesquelched) [inline docs/access.rst]:
```
``localhost:12345``
```

Comment (thesquelched) [inline docs/access.rst]: You know, I'm kind of turn on the double versus single backtick thing.  I like it for anything where monospace is an advantage, e.g. URL's, code, shell commands.  But on the other hand, there are some legimitate uses for single backticks, e.g. node names and such.  Do you think that's a good rule to go by?

Comment (jimbobhickville) [inline docs/access.rst]: I don't have a strong opinion.  After seeing the generated page, the single backtick isn't that noticeable.  It's just a very slight italics.

**Comment by thesquelched** [comment]
:+1:

**Comment by pchanne** [comment]
+1

**Pull request merged**